packages/components/src/Button/Button.types.ts

```diff
@@ -0,0 +1,126 @@
+import { IconNames } from "@jobber/design";
+import { LinkProps } from "react-router-dom";
+import { XOR } from "ts-xor";
+
+export type HTMLButtonType = "button" | "submit";
+export type ButtonVariation = "work" | "learning" | "subtle" | "destructive";
+export type ButtonSize = "small" | "base" | "large";
+export type ButtonType = "primary" | "secondary" | "tertiary";
+
+export interface ButtonFoundationProps {
+  /**
+   * Used for screen readers. Will override label on screen
+   * reader if present.
+   */
+  readonly ariaLabel?: string;
+  readonly ariaControls?: string;
+  readonly ariaHaspopup?: boolean;
+  readonly ariaExpanded?: boolean;
+  readonly disabled?: boolean;
+  readonly external?: boolean;
+  readonly fullWidth?: boolean;
+  readonly id?: string;
+  readonly loading?: boolean;
+  readonly size?: ButtonSize;
+  onClick?(
+    event: React.MouseEvent<HTMLAnchorElement | HTMLButtonElement>,
+  ): void;
+  onMouseDown?(
+    event: React.MouseEvent<HTMLAnchorElement | HTMLButtonElement>,
+  ): void;
+}
+
+interface ButtonNonComposableProps extends ButtonFoundationProps {
+  readonly icon?: IconNames;
+  readonly iconOnRight?: boolean;
+  readonly label?: string;
+}
+
+interface ButtonIconProps extends ButtonNonComposableProps {
+  readonly icon: IconNames;
+  readonly ariaLabel: string;
+}
+
+interface ButtonLabelProps extends ButtonNonComposableProps {
+  readonly label: string;
+}
+
+type BaseButtonProps = XOR<
+  BaseActionProps,
+  XOR<DestructiveActionProps, SubmitActionProps>
+> &
+  XOR<
+    XOR<SubmitButtonProps, BasicButtonProps>,
+    XOR<ButtonLinkProps, ButtonAnchorProps>
+  >;
+
+interface ButtonIconProps extends ButtonFoundationProps {
+  readonly icon: IconNames;
+  readonly ariaLabel: string;
+}
+
+interface ButtonLabelProps extends ButtonFoundationProps {
+  readonly label: string;
+}
+
+interface ButtonAnchorProps extends ButtonFoundationProps {
+  /**
+   * Used to create an 'href' on an anchor tag.
+   */
+  readonly url?: string;
+}
+
+interface ButtonLinkProps<S = unknown> extends ButtonFoundationProps {
+  /**
+   * **Deprecated**: to will be removed in the next major version
+   * @deprecated
+   */
+  readonly to?: LinkProps<S>["to"];
+}
+
+interface BaseActionProps extends ButtonFoundationProps {
+  readonly variation?: ButtonVariation;
+  readonly type?: ButtonType;
+}
+
+interface DestructiveActionProps extends ButtonFoundationProps {
+  readonly variation: Extract<ButtonVariation, "destructive">;
+  readonly type?: ButtonType;
+}
+
+interface SubmitActionProps
```

Comment:
Can we also include formId it would open the ability to place submit buttons outside of forms.